abstractions/typescript/src/authentication/accessTokenProvider.ts

```diff
@@ -0,0 +1,9 @@
+/**
+ * @interface
+ * An AccessTokenProvider implementation retrieves an access token
+ * to be used by an AuthenticationProvider implementation.
+ * @property {Function} authenticateRequest - The function to authenticate the request.
+ */
+ export interface AccessTokenProvider {
```

Comment:
why are we duplicating the work of #1039 here? Please revert those changes, and either rebase on that branch or wait for it to be merged.

Reply:
Testing. Please ignore the upcoming commits.  The scope of this PR is not going to change